hazelcast/src/test/java/com/hazelcast/cache/impl/CacheCreateUseDestroyTest.java

```diff
@@ -0,0 +1,223 @@
+        assumptions();
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(1);
+        HazelcastInstance member = factory.newHazelcastInstance(getConfig());
+        CachingProvider provider = Caching.getCachingProvider();
+        defaultCacheManager = provider.getCacheManager(null, null,
+                HazelcastCachingProvider.propertiesByInstanceItself(member));
+        cacheService = getNode(member).getNodeEngine().getService(ICacheService.SERVICE_NAME);
+        CacheEntryListenerFactory.listener = null;
+    }
+
+    protected void assumptions() {
+        Assume.assumeThat(inMemoryFormat, not(NATIVE));
+    }
+
+    @Override
+    protected Config getConfig() {
+        Config config = super.getConfig();
+        CacheSimpleEntryListenerConfig entryListenerConfig = new CacheSimpleEntryListenerConfig();
+        entryListenerConfig.setCacheEntryListenerFactory("com.hazelcast.cache.impl.CacheCreateUseDestroyTest$CacheEntryListenerFactory");
+        entryListenerConfig.setOldValueRequired(true);
+        entryListenerConfig.setSynchronous(true);
+        CacheSimpleConfig cacheSimpleConfig = new CacheSimpleConfig()
+                .setName("cache*")
+                .setInMemoryFormat(inMemoryFormat)
+                .setStatisticsEnabled(true)
+                .setManagementEnabled(true)
+                .setCacheEntryListeners(Collections.singletonList(entryListenerConfig));
+        if (inMemoryFormat == NATIVE) {
+            EvictionConfig evictionConfig = new EvictionConfig(90, USED_NATIVE_MEMORY_PERCENTAGE, EvictionPolicy.LFU);
+            cacheSimpleConfig.setEvictionConfig(evictionConfig);
+
+            NativeMemoryConfig memoryConfig = new NativeMemoryConfig();
+            memoryConfig.setEnabled(true);
+            memoryConfig.setSize(NATIVE_MEMORY_SIZE);
+            memoryConfig.setAllocatorType(NativeMemoryConfig.MemoryAllocatorType.STANDARD);
+            config.setNativeMemoryConfig(memoryConfig);
+        }
+
+        config.addCacheConfig(cacheSimpleConfig);
+        return config;
+    }
+
+    @Test
+    public void testCache_whenDestroyedByCacheManager() {
+        String cacheName = randomMapName("cache");
+        cache = defaultCacheManager.getCache(cacheName);
+        cache.put("key", "value");
+        cache.get("key");
+        assertCreatedCache(cacheName);
+
+        defaultCacheManager.destroyCache(cacheName);
+        assertDestroyedCache(cacheName);
+    }
+
+    @Test
+    public void testCache_whenDestroyedByICache_destroy() {
+        String cacheName = randomMapName("cache");
+        cache = defaultCacheManager.getCache(cacheName);
+        DistributedObject internalCacheProxy = cache.unwrap(DistributedObject.class);
+        cache.put("key", "value");
+        cache.get("key");
+        assertCreatedCache(cacheName);
+
+        internalCacheProxy.destroy();
+        assertDestroyedCache(cacheName);
+    }
+
+    private void assertDestroyedCache(String cacheName) {
```

Comment:
I would re-order `assertCreatedCache()` and `assertDestroyedCache()`, to ease the readability of the test.